File: frontend/src/application/ports/QualificationRepository.ts
import type { Qualification, QualificationKey } from '@/domain';

export interface QualificationRepository {
    findAll(): Promise<Qualification[]>;

    create(qualification: Qualification): Promise<Qualification>;

    update(qualificationKey: QualificationKey, qualification: Qualification): Promise<Qualification>;

    deleteByKey(qualificationKey: QualificationKey): Promise<void>;
}

File: backend/src/main/java/org/eventplanner/application/ports/QualificationRepository.java
package org.eventplanner.application.ports;

import java.util.List;
import java.util.Optional;

import org.eventplanner.domain.entities.Qualification;
import org.eventplanner.domain.values.QualificationKey;

public interface QualificationRepository {
    List<Qualification> findAll();

    Optional<Qualification> findByKey(QualificationKey qualificationKey);

    void create(Qualification qualification);

    void update(Qualification qualification);

    void deleteByKey(QualificationKey key);

    void deleteAll();
}
